litt/src/main.rs

```diff
@@ -20,131 +23,186 @@ use colored::*;
 
+impl fmt::Display for LittError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match &self {
+            LittError(s) => write!(f, "{}", s.red()),
+        }
+    }
+}
+
 fn main() -> Result<(), LittError> {
     let cli = Cli::parse();
 
-    let index_tracker =
-        IndexTracker::create(".litt".into()).map_err(|e| LittError(e.to_string()))?;
+    let mut index_tracker = match IndexTracker::create(".litt".into()) {
+        Ok(index_tracker) => index_tracker,
+        Err(e) => return Err(LittError(e.to_string())),
+    };
 
     // everything that does not require litt index
-    //
+
     // Print existing litt indices
     if cli.list {
         println!("Currently available indices:");
-        for index in index_tracker.all().map_err(|e| LittError(e.to_string()))? {
-            println!(" - {:?}", index);
+        match &index_tracker.all() {
+            Ok(indecies) => {
+                for index in indecies {
+                    println!(" - {:?}", index);
+                }
+            }
+            Err(e) => return Err(LittError(e.to_string())),
         }
         return Ok(());
     }
 
     // check if name of litt index was given by user
     if cli.litt_index.is_none() {
-        Cli::command()
-            .print_help()
-            .map_err(|e| LittError(e.to_string()))?;
+        if let Err(e) = Cli::command().print_help() {
+            return Err(LittError(e.to_string()));
+        }
         return Err(LittError("Litt index missing!".into()));
-    } else if let Some(index_name) = cli.litt_index {
-        // initialize new index
-        if !cli.init.is_empty() {
-            println!("Creating new index \"{}\" at: {}: ", index_name, cli.init);
-            if index_tracker.exists(&index_name) || index_tracker.path_exists(&cli.init).is_some() {
-                return Err(LittError(format!("Failed to create new index since a index at this path already exists: name: \"{}\", path: \"{}\"", index_tracker.get_name(&cli.init).unwrap_or_default(), cli.init)));
-            }
-            // Add new index to index tracker (adding first, so that it can be removed in case of
-            // failiure)
-            let start = Instant::now();
-            _ = index_tracker
-                .add(index_name, cli.init.clone())
-                .map_err(|e| LittError(e.to_string()));
-            // Create new index
-            let mut index = Index::create(&cli.init, SearchSchema::default())
-                .map_err(|e| LittError(e.to_string()))?;
-            index
-                .add_all_pdf_documents()
-                .map_err(|e| LittError(e.to_string()))?;
-            println!(
-                "Successfully indexed {} document pages in {:?}",
-                index.searcher().num_docs(),
-                start.elapsed()
-            );
-
-            return Ok(());
+    }
+
+    let index_name = cli
+        .litt_index
+        .expect("Unkown error when getting name from index");
```

Comment:
Even for unknown errors, I think we should have an extra enum variant that we return, instead of `expect`ing